src/persist/src/storage.rs

```diff
@@ -474,4 +520,44 @@ pub mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn check_meta_version() -> Result<(), Error> {
+        let registry = MemRegistry::new();
+        let mut blob = registry.blob_no_reentrance()?;
+
+        let meta = BlobMeta::default();
+        let mut val = Vec::new();
+        meta.encode(&mut val);
+        let current_version = val[0];
+
+        // This test needs to be able to increment and decrement the current
+        // version.
+        assert!(current_version > 0 && current_version < u8::MAX);
```

Comment:
nit: you could do `let (previous_version, future_version) = (current_version - 1, current_version + 1);` here and use those names below to minimize action at a distance